### PR TITLE
Updates

### PR DESCRIPTION
**Pull request by NiallRees**

### CODE REVIEW

File: tap_pinterest_ads/client.py [hunk 1]
```diff
@@ -61,3 +64,39 @@ def get_url_params(
     def parse_response(self, response: requests.Response) -> Iterable[dict]:
         """Parse the response and return an iterator of result rows."""
         yield from extract_jsonpath(self.records_jsonpath, input=response.json())
+
+    def validate_response(self, response: requests.Response) -> None:
```
Comment (NiallRees) [inline tap_pinterest_ads/client.py:68]: This logic applies to all endpoints from what I can see in the docs

File: tap_pinterest_ads/streams.py [hunk 2]
```diff
@@ -243,51 +240,78 @@ def get_url_params(
...
-            A decorated method.
-        """
-        decorator: Callable = backoff.on_exception(
-            backoff.expo,
-            (RetriableAPIError,),
-            max_tries=5,
-            factor=5,
-        )(func)
-        return decorator
+    def get_url_params(
```
Comment (NiallRees) [inline tap_pinterest_ads/streams.py:300]: We can definitely DRY this up with the above stream, but it works for now

File: tap_pinterest_ads/tap.py [hunk 3]
```diff
@@ -42,7 +44,7 @@
     th.Property(
         "start_date",
         th.DateTimeType,
-        required=True,
+        default="2019-10-17T00:00:00Z",
```
Comment (NiallRees) [inline tap_pinterest_ads/tap.py:47]: Added a default value here. This is the earliest value that the API lets you get some metrics:

> "Reporting for this date range cannot be broken down by conversion platform type (web, in-app, or offline). Either choose a date range that begins on October 17, 2019 or later or choose conversion platform columns which is allowed

File: tap_pinterest_ads/client.py (hunk 1 above)
Comment (GtheSheep) [reply tap_pinterest_ads/client.py:68]: yep, agreed!

**Review by GtheSheep — APPROVED** [review]
Thanks again!

**Pull request merged**